torch_frame/data/multi_embedding_tensor.py

```diff
@@ -98,29 +101,159 @@ def __getitem__(
+            dim (int): The dimension along which to narrow.
+            start (int): The starting dimension index to narrow.
+            length (int): The length of the dimension to narrow.
+
+        Returns:
+            MultiEmbeddingTensor: A :class:`MultiEmbeddingTensor` instance.
+
+        Example:
+            >>> num_rows = 2
+            >>> tensor_list = [
+            ...    torch.tensor([[0, 1, 2], [3, 4, 5]]),  # col0
+            ...    torch.tensor([[6, 7], [8, 9]]),        # col1
+            ...    torch.tensor([[10], [11]]),            # col2
+            ... ]
+            >>> out = MultiEmbeddingTensor.from_tensor_list(tensor_list)
+            >>> out
+            MultiEmbeddingTensor(num_rows=2, num_cols=3, device='cpu')
+            >>> out.narrow(dim=1, start=1, length=2)
+            MultiEmbeddingTensor(num_rows=2, num_cols=2, device='cpu')
+        """
+        dim = self._normalize_dim(dim)
+        num_data = self.num_rows if dim == 0 else self.num_cols
+        if start == 0 and start + length >= num_data:
+            return self
+        elif length <= 0:
+            return MultiEmbeddingTensor(
+                num_rows=0 if dim == 0 else self.num_rows,
+                num_cols=0 if dim == 1 else self.num_cols,
+                values=torch.tensor([], device=self.device),
+                offset=torch.tensor([0], device=self.device),
+            )
+
+        elif dim == 0:
+            return MultiEmbeddingTensor(
+                num_rows=length,
+                num_cols=self.num_cols,
+                values=self.values[start:start + length],
+                offset=self.offset,
+            )
+        elif dim == 1:
+            offset = self.offset[start:start + length + 1] - self.offset[start]
+            return MultiEmbeddingTensor(
+                num_rows=self.num_rows,
+                num_cols=length,
+                values=self.values[:, self.offset[start]:self.offset[start +
+                                                                     length]],
+                offset=offset,
+            )
+
+    def _slice(
+        self,
+        index: slice,
+        dim: int,
+    ) -> "MultiEmbeddingTensor":
+        dim = self._normalize_dim(dim)
+        num_data = self.num_rows if dim == 0 else self.num_cols
+        if index.step is not None and index.step > 1:
+            idx = torch.tensor(
+                range(num_data)[index],
+                device=self.device,
+                dtype=torch.long,
+            )
+            return self.index_select(idx, dim=dim)
+        else:
+            start_idx: int = self._normalize_index(index.start or 0, dim=dim)
+            end_idx: int = self._normalize_index(
+                index.stop or num_data,
+                dim=dim,
+                is_slice_end=True,
+            )
+            return self.narrow(
+                dim=dim,
+                start=start_idx,
+                length=end_idx - start_idx,
+            )
```

Comment:
Do you think `_slice`, `narrow`, `select`, `index_select` can be shared with `MultiNestedTensor` (in other words, we put them in the parent `_MultiTensor` class)? Here we only need to implement `_row_index_select`, `_col_index_select` and so on.

Reply:
Sounds good. I'll update the PR accordingly

Reply:
Actually, could I follow up on the refactor in a separate PR to first unblock #194 and #206? I'd like to keep this PR easy to review and not touch `MultiNestedTensor` if you're fine.

Reply:
Sounds good. Thanks.